### DISCUSSION TITLE
Watch vs getValues

### DISCUSSION THREAD

**Discussion by james-gardner** [react-hook-form/react-hook-form#1826] [Q&A] [answered]
Hi, 

Perhaps the name itself is what has got me confused but I'd like to understand what the fundamental difference is (if any) between ```watch``` and ```getValues```. In the documentation, there's a clear distinction but I don't understand what the different use cases of each are.

Could you clarify?

**Comment by bluebill1049** [comment, marked as answer]
- watch trigger re-render and listen changes for the inputs.
- getValues() is most used to read what's live in the current form.

**Reply by rezkesh** [reply]
So it I want to call an endpoint inside a callback function, should I use `getValues()` to have the latest values of the form?

**Reply by BrendanC23** [reply]
That's what I would recommend. `getValues()` will let you read all of the current form values.
